### DISCUSSION TITLE
NICE improvement. but some tasks could be done more automaticaly.

### DISCUSSION THREAD

**Discussion by osevan** [k4yt3x/wg-meshconf#28] [Ideas]
Thank you very much for your Python work, i don't tested your script yet,  but it looks very good. 

My additonal idea is auto deployment like wg-deploy command for automaticaly move configs to endpoints via ssh or scp and starting systemd service. 

I know its possible with bash,  but just for completion for autodeployment via ssh in combination wirh your project.

Imagine your admin notebook got ssh keys set already and can connect very well to your endpoints. You need only functions to start wg-deploy.

And i have question about dns addressing inside wg-meshconfig:

Why is dns needed and why you did not set  inside /etc/host file your dns name with endpoint ip address? 

Why you need dns inside wg-meshconfg database? 

Thanks and

Best regards

**Comment by k4yt3x** [comment]
Sorry for the late reply, just finished exams.

> My additonal idea is auto deployment like wg-deploy command for automaticaly move configs to endpoints via ssh or scp and starting systemd service.

I've considered that, it'll be something similar to Ansible. In fact, I think it'll be possible to just call Ansible to do this. I've just never had the motivation/time to work on it. It's going to take some work.

> Why is dns needed and why you did not set inside /etc/host file your dns name with endpoint ip address?

You mean why we need to use domain names for endpoints? It's just for the simplicity of demonstration. In practice, you can write it into hosts, or straight up use IP addresses. I think you could leave DNS blank iirc.

**Reply by osevan** [reply]
Thank you very much
